apps/server/src/classes/data-provider/DataProvider.utils.ts

```diff
@@ -27,3 +35,22 @@ export function safeMerge(existing: DatabaseModel, newData: Partial<DatabaseMode
     http: { ...existing.http, ...http },
   };
 }
+
+export function rundownToDatabaseRundown(newData: OntimeRundown): DatabaseOntimeRundown {
+  const databaseRundown = newData.flatMap((entry) => {
+    const entryClone = structuredClone(entry);
+    if (isOntimeEvent(entryClone)) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { delay, ...dbEvent } = entryClone;
+      return dbEvent satisfies DatabaseOntimeEvent as DatabaseOntimeEvent;
+    }
+    if (isOntimeBlock(entryClone)) {
+      return entryClone;
+    }
+    if (isOntimeDelay(entryClone)) {
+      return entryClone;
+    }
+  });
+
+  return databaseRundown;
+}
```

Comment:
todo: 
- we need to handle unknown types, maybe by just throwing
- we likely dont need the structured clone AND the destructuring on the same dataset
- we likely do not need the flatmap
tip: remember to turn on strict mode when you are working

maybe like


```suggestion
export function rundownToDatabaseRundown(newData: OntimeRundown): DatabaseOntimeRundown {
  const databaseRundown = newData.map((entry) => {
    if (isOntimeEvent(entry)) {
      // eslint-disable-next-line @typescript-eslint/no-unused-vars -- removing the delay from the event
      const { delay, ...dbEvent } = entry;
      return dbEvent as DatabaseOntimeEvent;
    } else if (isOntimeBlock(entry) || isOntimeDelay(entry)) {
      return { ...entry };
    } else {
      throw new Error('Unknown entry type');
    }
  });

  return databaseRundown;
}
```